hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/HapiApiClients.java

```diff
@@ -118,154 +111,75 @@ private ManagedChannel createNettyChannel(boolean useTls, final String host, fin
     }
 
     public TokenServiceBlockingStub getTokenSvcStub(AccountID nodeId, boolean useTls) {
-        return tokenSvcStubs.get(stubId(nodeId, useTls));
+        return randomStubsFromPool(channelPools.get(stubId(nodeId, useTls))).tokenSvcStubs();
     }
 
     public CryptoServiceBlockingStub getCryptoSvcStub(AccountID nodeId, boolean useTls) {
-        return cryptoSvcStubs.get(stubId(nodeId, useTls));
+        return randomStubsFromPool(channelPools.get(stubId(nodeId, useTls))).cryptoSvcStubs();
     }
 
     public FreezeServiceBlockingStub getFreezeSvcStub(AccountID nodeId, boolean useTls) {
-        return freezeSvcStubs.get(stubId(nodeId, useTls));
+        return randomStubsFromPool(channelPools.get(stubId(nodeId, useTls))).freezeSvcStubs();
     }
 
     public SmartContractServiceBlockingStub getScSvcStub(AccountID nodeId, boolean useTls) {
-        return scSvcStubs.get(stubId(nodeId, useTls));
+        return randomStubsFromPool(channelPools.get(stubId(nodeId, useTls))).scSvcStubs();
     }
 
-    public ConsensusServiceBlockingStub getConsSvcStub(
-            AccountID nodeId, boolean useTls, Set<HederaFunctionality> operations) {
-        final String id;
-        if (!operations.isEmpty()) {
-            // Currently we support only consensus operations.
-            // If we need to support other operations, then we should check
-            // the workflow operations have consensus operations
-            id = workflowStubId(nodeId, useTls);
-        } else {
-            id = stubId(nodeId, useTls);
-        }
-        return consSvcStubs.get(id);
+    public ConsensusServiceBlockingStub getConsSvcStub(AccountID nodeId, boolean useTls) {
+        return randomStubsFromPool(channelPools.get(stubId(nodeId, useTls))).consSvcStubs();
     }
 
     public NetworkServiceBlockingStub getNetworkSvcStub(AccountID nodeId, boolean useTls) {
-        return networkSvcStubs.get(stubId(nodeId, useTls));
+        return randomStubsFromPool(channelPools.get(stubId(nodeId, useTls))).networkSvcStubs();
     }
 
     public ScheduleServiceBlockingStub getScheduleSvcStub(AccountID nodeId, boolean useTls) {
-        return schedSvcStubs.get(stubId(nodeId, useTls));
+        return randomStubsFromPool(channelPools.get(stubId(nodeId, useTls))).scheduleSvcStubs();
     }
 
     public UtilServiceGrpc.UtilServiceBlockingStub getUtilSvcStub(AccountID nodeId, boolean useTls) {
-        return utilSvcStubs.get(stubId(nodeId, useTls));
+        return randomStubsFromPool(channelPools.get(stubId(nodeId, useTls))).utilSvcStubs();
     }
 
     private String stubId(AccountID nodeId, boolean useTls) {
         return useTls ? tlsStubIds.get(nodeId) : stubIds.get(nodeId);
     }
 
-    private String workflowStubId(AccountID nodeId, boolean useTls) {
-        return useTls ? workflowTlsStubIds.get(nodeId) : workflowStubIds.get(nodeId);
+    private static ChannelStubs randomStubsFromPool(@NonNull final List<ChannelStubs> stubs) {
```

Comment:
Just a thought here.  Random selection still leaves a roughly 1:50 chance we double up on a channel.  Would it be better to round-robin these so the probability of any given call being on the same channel as another is minimized?